infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironmentFactoryTest.java

```diff
@@ -297,6 +390,21 @@ public void exceptionOnRecipeLoadError() throws Exception {
     k8sEnvFactory.doCreate(internalRecipe, emptyMap(), emptyList());
   }
 
+  @Test(
+      expectedExceptions = ValidationException.class,
+      expectedExceptionsMessageRegExp =
+          "Environment can not contain two 'Service' objects with the same name 'db'")
+  public void exceptionOnObjectsWithTheSameNameAndKind() throws Exception {
+    HasMetadata object1 =
```

Comment:
well, different objects have own specific. 
`ConfigMap` - may be referenced as environment variables for containers or Volume for pod;
`Service` - should not be renamed because the user may rely on DNS;
`PVC` - references should be changed in Container;
etc.

`ConfigMap`, `PVC` may be handled on Che Server side, but think it is not critical for this PR. Will create a separate issue.

Reply:
:+1: